src/Elasticsearch.Net/Connection/RuntimeVersion.cs

```diff
@@ -0,0 +1,316 @@
+﻿// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+// Adapted from BenchmarkDotNet source https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/Environments/Runtimes/CoreRuntime.cs
+#region BenchmarkDotNet License https://github.com/dotnet/BenchmarkDotNet/blob/master/LICENSE.md
+// The MIT License
+// Copyright (c) 2013–2020.NET Foundation and contributors
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+// and associated documentation files (the "Software"), to deal in the Software without restriction,
+// including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all copies or substantial
+// portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT
+// LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#endregion
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+#if DOTNETCORE
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+#else
+using Microsoft.Win32;
+#endif
+
+namespace Elasticsearch.Net
+{
+	/// <summary>
+	/// Represents the current .NET Runtime version.
+	/// </summary>
+	internal sealed class RuntimeVersion
+	{
+		internal static readonly RuntimeVersion Default = new RuntimeVersion(new Version(0, 0, 0), false);
+
+		private RuntimeVersion(Version version, bool isPrerelease)
+		{
+			var finalVersion = version;
+
+			if (version.Minor == -1 || version.Build == -1)
```

Comment:
This should get flagged as a style violation in our .editorconfig. Be keen to find out why if it does not in VS. (e.g no {} on non complex statements.

Reply:
Will fix and investigate editor.config